source/iterator/pub_sub_listener_test.go

```diff
@@ -118,3 +120,74 @@ func TestNewCDCIterator(t *testing.T) {
 	assert.Equal(t, response.key, res.key)
 	assert.Equal(t, response.psc, res.psc)
 }
+
+func TestNewCDCIterator_Next(t *testing.T) {
+	redisChannel := "subchannel"
+	testMessage := "some_dummy_message"
+	testMessage1 := "some_dummy_message1"
+	mr, err := miniredis.Run()
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer mr.Close()
+	conn, err := redis.Dial("tcp", mr.Addr())
+	if err != nil {
+		t.Fatal(err)
+	}
+	response := PubSubIterator{
+		key:     redisChannel,
+		psc:     &redis.PubSubConn{Conn: conn},
+		records: []sdk.Record{},
+		mux:     &sync.Mutex{},
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	res, err := NewPubSubIterator(ctx, conn, redisChannel)
+	assert.Nil(t, err)
+	assert.NotNil(t, res)
+	assert.Equal(t, response.key, res.key)
+	assert.Equal(t, response.psc, res.psc)
+	// publish is a fire and forget method, give a few ms for goroutines to start
+	// otherwise messages might be lost and tests will fail
+	time.Sleep(10 * time.Millisecond)
+	mr.Publish(redisChannel, testMessage)
+	mr.Publish(redisChannel, testMessage1)
+
+	var rec sdk.Record
+	ticker := time.NewTicker(400 * time.Millisecond)
+	defer ticker.Stop()
+	retryCount := 0
+	for {
+		select {
+		case <-ctx.Done():
+		case <-ticker.C:
+			if retryCount >= 10 {
+				break
```

Comment:
Am I understanding this correctly: we are waiting for the message, and retrying up to ten times in case we get ErrBackoffRetry. If we the retry count is >= 10, we stop, and the test is successful?

Reply:
Thanks for pointing it out, Fixed